src/main/java/com/artipie/http/MainSlice.java

```diff
@@ -112,30 +114,29 @@ public MainSlice(
     }
 
     /**
-     * Metrics storage Yaml node.
+     * Checks that metrics are collected by Prometheus.
+     *
      * @param settings Artipie settings
-     * @return Yaml node, could be null
+     * @return True if metrics are collected by Prometheus.
      */
-    private static Optional<YamlMapping> metricsStorage(final Settings settings) {
-        return Optional.ofNullable(settings.meta().yamlMapping("metrics"))
-            .flatMap(metrics -> Optional.ofNullable(metrics.yamlMapping("storage")));
+    static boolean isPrometheusConfigAvailable(final Settings settings) {
+        return settings
+            .meta()
+            .yamlSequence("metrics")
+            .values()
+            .stream()
+            .filter(Objects::nonNull)
+            .map(YamlNode::asMapping)
+            .anyMatch(mapping -> MetricsFromConfig.PROMETHEUS.equals(mapping.string("type")));
     }
 
     /**
-     * Checks that metrics are collected by Prometheus.
+     * Metrics storage Yaml node.
      * @param settings Artipie settings
      * @return Yaml node, could be null
      */
-    private static boolean isPrometheusConfigAvailable(final Settings settings) {
-        final Optional<YamlMapping> myml =
-            Optional.ofNullable(settings.meta().yamlMapping("metrics"));
-        final boolean available;
-        if (myml.isPresent()) {
-            final String type = myml.get().string("type");
-            available = MetricsFromConfig.PROMETHEUS.equals(type);
-        } else {
-            available = false;
-        }
-        return available;
+    private static Optional<YamlMapping> metricsStorage(final Settings settings) {
+        return Optional.ofNullable(settings.meta().yamlMapping("metrics"))
```

Comment:
Once again, `metrics` is not a mapping, but a sequence. In this sequence item with type `asto` should be found and on this item `storage` section is present...